apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -429,6 +428,14 @@ read_cache(ResId) ->
         [] -> not_found
     end.
 
+read_manager_pid_from_cache(ResId) ->
+    case read_cache(ResId) of
+        not_found ->
+            erlang:error(badarg);
```

Comment:
nit: if we do `exit(noproc)` here, we can git rid of the `error:badarg` catch clause

Reply:
actually, we should add a test to cover this error instead.